common/include/NeuralNetworkBase.hpp

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include <TH1F.h>
+#include <TH2F.h>
+
+#include <UHH2/core/include/Hists.h>
+#include "UHH2/core/include/Event.h"
+
+typedef std::vector < std::vector < float > > Matrix2D;
+
+class NeuralNetworkBase {
+
+public:
+  explicit NeuralNetworkBase(const std::string& path_);
+  void extractLayers();
+  void extractVariables();
+  void PrintVariables();
+  void extractClasses();
+  std::vector<TString> GetClasses() {return classes;};
+  int GetClassIndex(TString key) {return std::distance(classes.begin(),std::find(classes.begin(), classes.end(), key) );};
+  void PrintClasses();
+  void extractNormalization();
+  void PrintNormalization();
+  std::vector< float > readLineWeights(TString token);
+  void LoadLayers();
+  void PrintLayers();
+  void NormalizeInput();
+  void Apply(const TopJet& jet);
+
+  void CreateInputs(const TopJet& jet);
+
+  Matrix2D MatrixMultiplication( Matrix2D A, Matrix2D B);
+  Matrix2D MatrixAddition( Matrix2D A, Matrix2D B);
+  Matrix2D ApplyActivation( Matrix2D A, TString activation);
+  Matrix2D getOutputs() {return outputs;};
+
+protected:
+
+  TString path;
+  TString fname = path+"mymodel.txt";
```

Comment:
Might be good idea to have a "/" preceeding each filename? e.g . `path+"/mymodel.txt"` so that the user doesn't have to worry about a missing trailing `/` in `path`? I guess one would then also need to check if `path == ""` then set `path="."` (or similar).